Assets/Scripts/Layers/LayerTypes/GeoJSONPointLayer.cs

```diff
@@ -125,36 +127,34 @@ private Material GetMaterialInstance()
         /// to remove visualisations that are out of view
         /// </summary>
         public void RemoveFeaturesOutOfView()
-        {         
+        {
             // Remove visualisations that are out of view
             var frustumPlanes = GeometryUtility.CalculateFrustumPlanes(Camera.main);
-            for (int i = SpawnedVisualisations.Count - 1; i >= 0 ; i--)
+            foreach (var kvp in spawnedVisualisationDictionary)
             {
-                // Make sure to recalculate bounds because they can change due to shifts
-                SpawnedVisualisations[i].CalculateBounds();
-
-                var inCameraFrustum = GeometryUtility.TestPlanesAABB(frustumPlanes, SpawnedVisualisations[i].bounds);
+                var visualisation = kvp.Value;
+                visualisation.CalculateBounds();
+                var inCameraFrustum = GeometryUtility.TestPlanesAABB(frustumPlanes, visualisation.bounds);
                 if (inCameraFrustum)
                     continue;
 
-                var featureVisualisation = SpawnedVisualisations[i];
-                RemoveFeature(featureVisualisation);
+                RemoveFeature(kvp);
```

Comment:
Dit is niet goed, je krijgt zo een collection modified exception. De reden dat er een reversed for loop was is omdat je dan wel de collection kan modifien. je removed nu een entry uit de keyvaluepair terwijl hij potentieel nog niet klaar is met loopen